### PR TITLE
applications: asset_tracker_v2: Change to deferred logging

### PR DESCRIPTION
**Pull request by simensrostad**
Immediate log mode impacts the performace of the system because the
logging is being executed in context of the caller. Due to this some
time critical functions can fail if a lot of logging is enabled.
For instance when the debug configuration `overlay-debug.conf`
is enabled. This patch changes back to deferred logging to alleviate
this issue.

Deferred logging was previously reverted to immediate due to floats
not being properly logged. This has now been fixed.

**edit**
In addition, logging for nRF Cloud A/P-GPS libraries are turned of to
reduce the stress on the logger. Without this config the user still get
enough information from the application to indicate reason of failure,
if any.

Signed-off-by: Simen S. Røstad <simen.rostad@nordicsemi.no>

### CODE REVIEW

**Review by jtguggedal — APPROVED** [review]

**Review by coderbyheart — APPROVED** [review]

File: applications/asset_tracker_v2/overlay-debug.conf [hunk 1]
```diff
@@ -16,6 +16,9 @@
 # 4 DEBUG, maximal level set to LOG_LEVEL_DBG
 CONFIG_LOG_MAX_LEVEL=4
 
+# Increase log buffer size to prevent logs being dropped.
+CONFIG_LOG_BUFFER_SIZE=2304
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/overlay-debug.conf:20]: Any reason for this number? May be it is better to set to 2048?

**Review by balaji-nordic — APPROVED** [review]
Looks good. Just one question needs to be addressed.

File: applications/asset_tracker_v2/overlay-debug.conf (hunk 1 above)
Comment (simensrostad) [reply applications/asset_tracker_v2/overlay-debug.conf:20]: Yes, 2048 wasn't enough :)

**Comment by simensrostad** [comment]
Removed logging for nRF Cloud A/P-GPS handling. See updated commit msg / PR description.

**Review by balaji-nordic — APPROVED** [review]

**Pull request merged**